FlowCrypt/Functionality/Services/Folders Services/FoldersService.swift

```diff
@@ -53,18 +53,22 @@ final class FoldersService: FoldersServiceType {
         let fetchedFolders = try await self.remoteFoldersProvider.fetchFolders()
         return try await withCheckedThrowingContinuation { continuation in
             DispatchQueue.main.async {
-                // TODO: - Ticket? - instead of removing all folders remove only
-                // those folders which are in DB and not in remoteFolders
-                self.localFoldersProvider.removeFolders(for: user.email)
+                do {
+                    // TODO: - Ticket? - instead of removing all folders remove only
+                    // those folders which are in DB and not in remoteFolders
+                    try self.localFoldersProvider.removeFolders(for: user.email)
```

Comment:
@tomholub
Maybe you need this transactional? Now this are independent DB changes.

Reply:
Maybe, you can use your judgement